src/som/interpreter/actors/SPromise.java

```diff
@@ -94,6 +106,29 @@ public final boolean isValue() {
     return false;
   }
 
+  /**
+   * Do not use for things other than serializing Promises.
+   *
+   * @return the value this promise was resolved to
+   */
+  public final Object getValue() {
```

Comment:
```suggestion
  public final Object getValueForSnapshot() {
```

I would try to be as explicit as possible.
Also, there is already an `getValueUnsync()` method, which would make me assume that you likely need to have a `synchronized` in here. You have race condition here at the moment.